addons/im_livechat/models/chatbot_script.py

```diff
@@ -0,0 +1,201 @@
+    livechat_channel_count = fields.Integer(string='Livechat Channel Count', compute='_compute_livechat_channel_count')
+    first_step_warning = fields.Selection([
+        ('first_step_operator', 'First Step Operator'),
+        ('first_step_invalid', 'First Step Invalid'),
+    ], compute="_compute_first_step_warning")
+
+    def _compute_livechat_channel_count(self):
+        channels_data = self.env['im_livechat.channel.rule'].read_group(
+            [('chatbot_script_id', 'in', self.ids)], ['channel_id:count_distinct'], ['chatbot_script_id'])
+        mapped_channels = {channel['chatbot_script_id'][0]: channel['channel_id'] for channel in channels_data}
+        for script in self:
+            script.livechat_channel_count = mapped_channels.get(script.id, 0)
+
+    @api.depends('script_step_ids.step_type')
+    def _compute_first_step_warning(self):
+        for script in self:
+            allowed_first_step_types = [
+                'question_selection',
+                'question_email',
+                'question_phone',
+                'free_input_single',
+                'free_input_multi',
+            ]
+            welcome_steps = script.script_step_ids and script._get_welcome_steps()
+            if welcome_steps and welcome_steps[-1].step_type == 'forward_operator':
+                script.first_step_warning = 'first_step_operator'
+            elif welcome_steps and welcome_steps[-1].step_type not in allowed_first_step_types:
+                script.first_step_warning = 'first_step_invalid'
+            else:
+                script.first_step_warning = False
+
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        """ Correctly copy the 'triggering_answer_ids' field from the original script_step_ids to the clone.
+        This needs to be done in post-processing to make sure we get references to the newly created
+        answers from the copy instead of references to the answers of the original.
+
+        This implementation assumes that the order of created steps and answers will be kept between
+        the original and the clone, using 'zip()' to match the records between the two. """
+
+        default = default or {}
+        default['title'] = self.title + _(' (copy)')
+
+        clone_chatbot_script = super().copy(default=default)
+        if 'question_ids' in default:
+            return clone_chatbot_script
+
+        answers_map = {
+            original_answer: clone_answer
+            for clone_answer, original_answer
+            in zip(clone_chatbot_script.script_step_ids.answer_ids, self.script_step_ids.answer_ids)
+        }
+
+        for clone_step, original_step in zip(clone_chatbot_script.script_step_ids, self.script_step_ids):
+            clone_step.write({
+                'triggering_answer_ids': [
+                    (4, answer.id)
+                    for answer in [
+                        answers_map.get(original_answer)
```

Comment:
Is there any situation where it would make sense to have the original answer be missing here? If not, it should probably use a normal getitem.